gradle-plugin/src/test/kotlin/com/google/devtools/ksp/gradle/processor/TestSymbolProcessor.kt

```diff
@@ -19,8 +19,9 @@ package com.google.devtools.ksp.gradle.processor
 import com.google.devtools.ksp.processing.CodeGenerator
 import com.google.devtools.ksp.processing.KSPLogger
 import com.google.devtools.ksp.processing.SymbolProcessor
+import com.google.devtools.ksp.processing.SymbolProcessorProvider
 
-abstract class TestSymbolProcessor : SymbolProcessor {
+abstract class TestSymbolProcessor : SymbolProcessor, SymbolProcessorProvider {
```

Comment:
Can you convert this processor into the new model you proposed where `var`s for stuff like `logger` and `codeGenerator` is converted to `val`s? Same for the `init-plus-provider` integration test.

Reply:
@neetopia, I updated the branch and converted `TestSymbolProcessor` to the new model in [this commit](https://github.com/edrd-f/ksp/pull/1/commits/9fd2db07314437a2acce9d63080f71b14f0476af). 

About the `init-plus-provider` integration test, I'm not sure I understand what you mean. The test uses both models, so the one with `init` uses `lateinit`, naturally.